facilitator/src/idl.rs

```diff
@@ -1087,35 +1094,139 @@ impl Packet for InvalidPacket {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ValidationBatch {
+    pub sig: BatchSignature,
+    pub header: Vec<u8>,
+    pub packets: Vec<u8>,
+}
+
+impl ValidationBatch {
+    /// Reads and parses one Header from the provided std::io::Read instance.
+    pub fn read<R: Read>(reader: R) -> Result<Self, Error> {
+        let mut reader = Reader::with_schema(*VALIDATION_BATCH_SCHEMA, reader)
+            .map_err(|e| Error::AvroError("failed to create Avro reader".to_owned(), e))?;
+        // We expect exactly one record and for it to be an ingestion signature
+        let value = match reader.next() {
+            Some(Ok(value)) => value,
+            Some(Err(e)) => {
+                return Err(Error::AvroError(
+                    "failed to read record from Avro reader".to_owned(),
+                    e,
+                ));
+            }
+            None => return Err(Error::EofError),
+        };
+        if reader.next().is_some() {
+            return Err(Error::MalformedBatchError(
+                "excess value in reader".to_owned(),
+            ));
+        }
+
+        ValidationBatch::try_from(value)
+    }
+
+    /// Serializes this message into Avro format and writes it to the provided
+    /// std::io::Write instance.
+    pub fn write<W: Write>(&self, writer: &mut W) -> Result<(), Error> {
+        let mut writer = Writer::new(*VALIDATION_BATCH_SCHEMA, writer);
+        writer.append(Value::from(self)).map_err(|e| {
+            Error::AvroError("failed to append record to Avro writer".to_owned(), e)
+        })?;
+        writer
+            .flush()
+            .map_err(|e| Error::AvroError("failed to flush".to_owned(), e))?;
+        Ok(())
+    }
+}
+
+impl TryFrom<Value> for ValidationBatch {
+    type Error = Error;
+
+    fn try_from(value: Value) -> Result<Self, Self::Error> {
+        let fields = match value {
+            Value::Record(fields) => fields,
+            _ => {
+                return Err(Error::MalformedBatchError(
+                    "value is not a Record".to_owned(),
+                ))
+            }
+        };
+
+        let mut sig = None;
+        let mut header = None;
+        let mut packets = None;
+
+        for (field_name, field_value) in fields {
+            match (field_name.as_str(), field_value) {
+                ("sig", field_value) => sig = Some(BatchSignature::try_from(field_value)?),
+                ("header", Value::Bytes(buf)) => header = Some(buf),
+                ("packets", Value::Bytes(buf)) => packets = Some(buf),
+                _ => (),
```

Comment:
I think we should error out if we encounter any unexpected fields.

Reply:
See response to the other instance of this comment--I think this is what we want, but I'll revert if you disagree even w/ my reasoning.